bin/run_demo.sh

```diff
@@ -83,76 +83,77 @@ pipetask run -d "exposure=903342 AND detector=10" -b DATA_REPO/butler.yaml \
     --register-dataset-types -p "$pipeline#isr" \
     --instrument lsst.obs.subaru.HyperSuprimeCam --output demo_collection2 --replace-run
 
-# Test the execution butler.
-# Create execution butler with new graph and output collection.
+# Run the execution butler in multiple steps, ensuring that a fresh
+# butler is used each time.
+refresh_butler() {
+  rm -rf "$1"
+  mkdir "$1"
+  cp "$exedir"/* "$1/"
+}
 
-graph_file="test_exe.qgraph"
-exedir="./execution_butler"
-# This collection name must match that used in the Python tests.
-exeoutput="demo_collection_exe"
-exerun="$exeoutput/YYYYMMDD"
+test_execution_butler() {
+  # Test the execution butler.
+  # Create execution butler with new graph and output collection.
 
-pipetask qgraph -b DATA_REPO/butler.yaml \
+  graph_file="test_exe.qgraph"
+  exedir="./execution_butler"
+  # This collection name must match that used in the Python tests.
+  exeoutput="demo_collection_exe"
+  exerun="$exeoutput/YYYYMMDD"
+
+  pipetask qgraph -b DATA_REPO/butler.yaml \
     --input "$incoll" \
     -p "$pipeline" \
     -q "$graph_file" \
     --save-execution-butler "$exedir" \
     --clobber-execution-butler \
     --instrument lsst.obs.subaru.HyperSuprimeCam --output "$exeoutput" --output-run "$exerun"
 
-# Run the execution butler in multiple steps, ensuring that a fresh
-# butler is used each time.
-
-tmp_butler="./tmp_execution_butler"
-refresh_exe() {
-  rm -rf "$tmp_butler"
-  mkdir "$tmp_butler"
-  cp "$exedir"/* "$tmp_butler/"
+  # Run the init step
+  TMP_BUTLER="./tmp_execution_butler"
+  refresh_butler $TMP_BUTLER
+  pipetask --long-log run -b "$TMP_BUTLER" -i "$incoll" --output-run "$exerun" --init-only --register-dataset-types --qgraph "$graph_file" --extend-run
+
+  # Run the three quanta one at a time to ensure that we can start from a
```

Comment:
We are not running serially "to ensure" that.  The original comment had a comma to indicate that the fresh butler was used each time. How about:

```
# Run the three quanta sequentially with a refreshed execution butler ever time.
```